### PR TITLE
This adds the remove-local-config-resources function to the catalog

### PR DESCRIPTION
**Pull request by g-awmalik**
Initial commit for the remove-annotated-resources function.

### CODE REVIEW

File: functions/go/remove-annotated-resources/main.go
Comment (mengqiy) [reply functions/go/remove-annotated-resources/main.go:31]: Adding that would be a better UX.

File: functions/go/remove-annotated-resources/README.md
Comment (g-awmalik) [reply functions/go/remove-annotated-resources/README.md:1]: Acked

File: functions/go/remove-annotated-resources/metadata.yaml
Comment (g-awmalik) [reply functions/go/remove-annotated-resources/metadata.yaml:4]: Added

File: functions/go/remove-annotated-resources/go.mod
Comment (g-awmalik) [reply functions/go/remove-annotated-resources/go.mod:4]: Acked

File: functions/go/remove-annotated-resources/go.mod
Comment (g-awmalik) [reply functions/go/remove-annotated-resources/go.mod:6]: Acked

File: functions/go/remove-annotated-resources/README.md
Comment (g-awmalik) [reply functions/go/remove-annotated-resources/README.md:38]: Acked

File: functions/go/remove-annotated-resources/README.md
Comment (g-awmalik) [reply functions/go/remove-annotated-resources/README.md:78]: Acked

File: functions/go/remove-annotated-resources/README.md
Comment (g-awmalik) [reply functions/go/remove-annotated-resources/README.md:7]: Acked

File: functions/go/remove-annotated-resources/README.md
Comment (g-awmalik) [reply functions/go/remove-annotated-resources/README.md:12]: Acked

File: examples/remove-annotated-resources-simple/README.md
Comment (g-awmalik) [reply examples/remove-annotated-resources-simple/README.md:6]: Acked.

File: functions/go/remove-annotated-resources/main.go
Comment (g-awmalik) [reply functions/go/remove-annotated-resources/main.go:31]: I've added results to show how many resources were pruned and their names.

File: functions/go/remove-annotated-resources/main.go
Comment (g-awmalik) [reply functions/go/remove-annotated-resources/main.go:22]: Added a unit test.

File: functions/go/remove-local-config-resources/main.go
Comment (bharathkkb) [inline functions/go/remove-local-config-resources/main.go:62]: 1 result per resource maybe better?

File: functions/go/remove-local-config-resources/main.go
Comment (bharathkkb) [inline functions/go/remove-local-config-resources/main.go:57]: If none pruned it makes to add this result as a warning

File: functions/go/remove-local-config-resources/main.go
Comment (bharathkkb) [inline functions/go/remove-local-config-resources/main.go:75]:
```suggestion
			Message:  fmt.Sprintf("failed to remove local configs: %s", errMsg),
```

File: functions/go/remove-local-config-resources/main_test.go
Comment (bharathkkb) [inline functions/go/remove-local-config-resources/main_test.go:50]: require.NoError() maybe cleaner

File: functions/go/remove-local-config-resources/main_test.go
Comment (bharathkkb) [inline functions/go/remove-local-config-resources/main_test.go:59]: debug?

File: functions/go/remove-local-config-resources/main_test.go
Comment (bharathkkb) [inline functions/go/remove-local-config-resources/main_test.go:65]:
```suggestion
			require.Equal(t, resultMessage, test.ExpectedResult)
```

File: functions/go/remove-local-config-resources/main_test.go
Comment (bharathkkb) [inline functions/go/remove-local-config-resources/main_test.go:70]: Pattern I have seen is to have yamls in code, but I'll defer to the maintainers

File: functions/go/remove-local-config-resources/main_test.go
Comment (bharathkkb) [inline functions/go/remove-local-config-resources/main_test.go:8]: commented code

File: functions/go/remove-local-config-resources/main.go
Comment (bharathkkb) [inline functions/go/remove-local-config-resources/main.go:53]: Do we need both? Isn't len(prunedNames) == prunedCount?

File: functions/go/remove-local-config-resources/main.go
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/main.go:62]: Acked

File: functions/go/remove-local-config-resources/main.go
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/main.go:57]: You're right. Coding past 10pm will do that to you.

File: functions/go/remove-local-config-resources/main.go
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/main.go:75]: Acked

File: functions/go/remove-local-config-resources/main_test.go
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/main_test.go:59]: Removed

File: functions/go/remove-local-config-resources/main_test.go
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/main_test.go:8]: removed

File: functions/go/remove-local-config-resources/main.go
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/main.go:53]: Nope you don't. Coding past 10pm will do that to you.

File: functions/go/remove-local-config-resources/testdata/applied.yaml
Comment (mengqiy) [inline functions/go/remove-local-config-resources/testdata/applied.yaml:6]: Please always add a newline at the end of a file.
Same for other files below.

File: examples/remove-local-config-resources-simple/README.md
Comment (mengqiy) [inline examples/remove-local-config-resources-simple/README.md:1]: Only one `#` please

Comment (mengqiy) [reply examples/remove-local-config-resources-simple/README.md:1]: I'd suggest to name it `remove-local-config-resources: Imperative Example`.

File: functions/go/remove-local-config-resources/go.mod
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/go.mod:11]: I did and there's no difference to the `go.mod` file

File: examples/remove-local-config-resources-simple/.expected/diff.patch
Comment (g-awmalik) [reply examples/remove-local-config-resources-simple/.expected/diff.patch:20]: acked

File: functions/go/remove-local-config-resources/README.md
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/README.md:18]: Added

File: functions/go/remove-local-config-resources/README.md
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/README.md:23]: Added

File: functions/go/remove-local-config-resources/README.md
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/README.md:37]: Acked

File: functions/go/remove-local-config-resources/main.go
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/main.go:63]: Added. Although, the `internal.config.kubernetes.io/path` keeps coming up as empty.

File: functions/go/remove-local-config-resources/main.go
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/main.go:58]: Acked.

File: functions/go/remove-local-config-resources/main.go
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/main.go:88]: I added another field i.e. `path` to it. I think its OK to keep it now.

File: functions/go/remove-local-config-resources/testdata/applied.yaml
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/testdata/applied.yaml:6]: Acked

File: functions/go/remove-local-config-resources/main.go
Comment (natasha41575) [reply functions/go/remove-local-config-resources/main.go:63]: It's currently `config.kubernetes.io/path` - we still need to migrate this to match the function spec. Will do that soon.

File: examples/remove-local-config-resources-simple/README.md
Comment (g-awmalik) [reply examples/remove-local-config-resources-simple/README.md:1]: Changed. And renamed the folder as well.

File: functions/go/remove-local-config-resources/README.md
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/README.md:12]: That makes sense. I've. changed accordingly.

File: functions/go/remove-local-config-resources/README.md
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/README.md:93]: I thought it was required as part of the template. I've removed it.

File: functions/go/remove-local-config-resources/main.go
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/main.go:36]: Acked.

File: functions/go/remove-local-config-resources/main.go
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/main.go:46]: Great point. Done.

File: functions/go/remove-local-config-resources/main.go
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/main.go:72]: Cool. I've added a check to pull it from internal.* if that shows up empty, i'll pull it from config.*

File: functions/go/remove-local-config-resources/main.go
Comment (mengqiy) [inline functions/go/remove-local-config-resources/main.go:55]: Instead of creating a `Result` object here. You can directly construct `framework.ResultItem`.
You can eliminate the for loop in line 71. We can reduce the size of this file.

File: functions/go/remove-local-config-resources/README.md
Comment (mengqiy) [inline functions/go/remove-local-config-resources/README.md:24]: I'd suggest move these steps to the top of the Usage section.
So the usage section has the following order:
1) steps how it works.
2) Imperative
3) declarative

File: functions/go/remove-local-config-resources/main.go
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/main.go:55]: Good idea. I've made that change.

File: functions/go/remove-local-config-resources/README.md
Comment (g-awmalik) [reply functions/go/remove-local-config-resources/README.md:18]: Done.

**Review by mengqiy — APPROVED** [review]
LGTM

**Pull request merged**
